src/core/src/op/loop.cpp

```diff
@@ -191,6 +191,66 @@ void op::v5::Loop::validate_and_infer_types() {
     // Body
     m_bodies[0]->validate_nodes_and_infer_types();
 
+    // if output shape is not same with input in a back-edge, here will update the input shape
+    // Port map processing: output -> input
+    std::map<uint64_t, uint64_t> back_edges;
+    for (const auto& desc : get_input_descriptions()) {
+        if (const auto& merged_desc = std::dynamic_pointer_cast<opset5::Loop::MergedInputDescription>(desc)) {
+            back_edges[merged_desc->m_body_value_index] = merged_desc->m_body_parameter_index;
+        }
+    }
+    if (!back_edges.empty()) {
+        bool need_reinvalidate = false;
+        for (const auto& output_description : m_output_descriptions[0]) {
+            auto body_value = m_bodies[0]->get_results().at(output_description->m_body_value_index)->input_value(0);
+
+            if (auto body_output_description =
+                    ov::as_type_ptr<v0::TensorIterator::BodyOutputDescription>(output_description)) {
+                const ov::PartialShape& body_value_shape = body_value.get_partial_shape();
+                if (body_value_shape.rank().is_static()) {
+                    // handle the case: when sub-model's output shape does not compatible input shape in a back-edge,
+                    // such as
+                    //          Parameter(out:-1, 1)->|
+                    //                                |->Concat(out:-1, 2)->Result(out:-1, 2)
+                    //          Parameter(out:-1, 1)->|   (axis==1)
+                    // when iteration number is unknown or sub-model output shape may be vary, the Result shape should
+                    // infer as (-1, -1), then set changed one to input and propagate to others.
+                    if (back_edges.count(output_description->m_body_value_index)) {
+                        auto input_param =
+                            m_bodies[0]->get_parameters().at(back_edges[output_description->m_body_value_index]);
+                        const auto& input_param_ps = input_param->get_partial_shape();
+                        if (input_param_ps.rank().is_static()) {
+                            const auto body_rank_len = body_value_shape.rank().get_length();
+                            const auto input_rank_len = input_param_ps.rank().get_length();
+                            // if input and output shape are {n} or {} and not compatible, make output shape dynamic
+                            if (body_rank_len <= 1 && input_rank_len <= 1 &&
+                                !input_param_ps.compatible(body_value_shape)) {
+                                // reset sub model input shape
+                                input_param->set_partial_shape({-1});
+                                need_reinvalidate = true;
+                            } else if (body_rank_len == input_rank_len &&
```

Comment:
what about  `body_rank_len != input_rank_len` case? is it handled?

Reply:
If `body_rank_len != input_rank_len` occurs just follow the original path. There is already [a case](https://github.com/luo-cheng2021/openvino/blob/ef7017cf8c85c0685e61fef2c3fbe6d76a77c17f/src/core/tests/type_prop/loop.cpp#L351) that the Result of backedge's rank is 4 and Parameter's rank is 1.

Reply:
Added handling method for `body_rank_len != input_rank_len`. The [original testcase](https://github.com/luo-cheng2021/openvino/blob/ef7017cf8c85c0685e61fef2c3fbe6d76a77c17f/src/core/tests/type_prop/loop.cpp#L351) will change the condition rank to 4 which could not satisfy the spec.

Reply:
I think we still need more modifications. Please read my new bunch of comments.